### PR TITLE
[TASK] Improve warnings and debug info when references are not found.

### PR DESCRIPTION
**Pull request by linawolf**
Also apply spaceless to the link and reference templates.

Suggest probable targets by Levenstein distance

### CODE REVIEW

File: tests/Integration/tests/body-elements/line-blocks/expected/index.html [hunk 1]
```diff
@@ -1,38 +1,17 @@
...
-                    are preserved.
-                </div>
-            </div>
-            <div class="line">
-                Continuation lines are wrapped
-                portions of long lines; they begin
-                with spaces in place of vertical bars.
-            </div>
-        </div>
+            <p>| Line blocks are useful for addresses,
```
Comment (garvinhicking) [inline tests/Integration/tests/body-elements/line-blocks/expected/index.html:5]: Shouldn't this remain using `<div>` output style tags?

**Review by garvinhicking — COMMENTED** [review]
I love the extra debugging 🥇 
Can't comment on Levenshtein, I don't know him well. ;)

File: tests/Integration/tests/body-elements/line-blocks/expected/index.html (hunk 1 above)
Comment (linawolf) [reply tests/Integration/tests/body-elements/line-blocks/expected/index.html:5]: arg true. Copying real output into expected should exclude unrelated skipped tests

File: packages/guides/src/ReferenceResolvers/DocReferenceResolver.php [hunk 2]
```diff
@@ -29,11 +34,27 @@ public function resolve(LinkInlineNode $node, RenderContext $renderContext): boo
...
+                sprintf(
+                    'Document with name "%s" not found, required in file "%s". %s',
+                    $canonicalDocumentName,
+                    $renderContext->getCurrentFileName(),
+                    $didYouMean,
+                ),
+                array_merge($renderContext->getLoggerInformation(), $node->getDebugInformation()),
+            );
+
+            return true;
```
Comment (jaapio) [inline packages/guides/src/ReferenceResolvers/DocReferenceResolver.php:57]: Should this not be false? As there is nothing resolved. 

Maybe we should move the suggestions to a different resolver with the lowest priority, So the chain would be:

- DocReference
- Interlink
- ....
- SuggestionDocs
- SuggestionInterlink

File: packages/guides/src/ReferenceResolvers/InterlinkReferenceResolver.php [hunk 3]
```diff
@@ -27,18 +33,55 @@ public function resolve(LinkInlineNode $node, RenderContext $renderContext): boo
...
+            $this->logger->warning(
+                sprintf(
+                    'Inventory with name "%s" could not be resolved in file "%s". ',
+                    $domain,
+                    $renderContext->getCurrentFileName(),
+                ),
+                array_merge($renderContext->getLoggerInformation(), $node->getDebugInformation()),
+            );
+
+            return true;
```
Comment (jaapio) [inline packages/guides/src/ReferenceResolvers/InterlinkReferenceResolver.php:45]: Same here, should this not be false?

File: packages/guides/src/ReferenceResolvers/InterlinkReferenceResolver.php [hunk 4]
```diff
@@ -27,18 +33,55 @@ public function resolve(LinkInlineNode $node, RenderContext $renderContext): boo
...
+                sprintf(
+                    'Inventory with name "%s" does not contain group %s, required in file "%s". ',
+                    $domain,
+                    $group,
+                    $renderContext->getCurrentFileName(),
+                ),
+                array_merge($renderContext->getLoggerInformation(), $node->getDebugInformation()),
+            );
+
+            return true;
```
Comment (jaapio) [inline packages/guides/src/ReferenceResolvers/InterlinkReferenceResolver.php:61]: Same here, should this not be false?

File: packages/guides/src/ReferenceResolvers/InterlinkReferenceResolver.php [hunk 5]
```diff
@@ -27,18 +33,55 @@ public function resolve(LinkInlineNode $node, RenderContext $renderContext): boo
...
+                    $domain,
+                    $target,
+                    $group,
+                    $renderContext->getCurrentFileName(),
+                    $didYouMean,
+                ),
+                array_merge($renderContext->getLoggerInformation(), $node->getDebugInformation()),
+            );
+
+            return true;
```
Comment (jaapio) [inline packages/guides/src/ReferenceResolvers/InterlinkReferenceResolver.php:84]: Same here, should this not be false?

**Review by jaapio — COMMENTED** [review]
I see that you repeated the suggestion algorithm multiple times. It feels a bit strange to have it spread over the code as it is part of our resolver logic. 

A suggestion to move it to a separate resolver, which is aware of multiple strategies? or maybe multiple separate resolvers? 

I think the changes of `return false` to `return true` might cause issues, so please let's talk about that why it's required to have this change.

File: packages/guides/src/ReferenceResolvers/DocReferenceResolver.php (hunk 2 above)
Comment (linawolf) [reply packages/guides/src/ReferenceResolvers/DocReferenceResolver.php:57]: If we would do that we would output multiple warnings for the same reference?

**Comment by linawolf** [comment]
Agreed with @jaapio that this PR needs more though...

**Comment by linawolf** [comment]
Gonna split this up into smaller, reviewable parts and apply suggestions

**Pull request closed without merging**